### ISSUE TITLE
How did the old cocoadialog get removed?

### ISSUE THREAD

**Issue by heisian** [cocoadialog/cocoadialog#109]
Did this repo get force-pushed or something removing the old release?

Related homebrew (since removed) script:
```
cask 'cocoadialog' do
  version '2.1.1'
  sha256 '7f721d221697381a51df6091153b0470ce09f4083d86bbb93446db88eac58827'

  # github.com/downloads/mstratman/cocoadialog was verified as official when first introduced to the cask
  url "https://github.com/downloads/mstratman/cocoadialog/CocoaDialog-#{version}.dmg"
  appcast 'https://github.com/mstratman/cocoadialog/releases.atom',
          checkpoint: '7144c245f8616fcadd57144910fec921871a3af7478fe4b62eab9f5f89501aeb'
  name 'cocoaDialog'
  homepage 'https://mstratman.github.io/cocoadialog/'

  app 'CocoaDialog.app'
end
```

**Comment by gingerbeardman** [comment]
I think it was moved to an org and some things fell by the wayside.

**Comment by prohtex** [comment]
Can we get an update on this project? Beyond the splash page and donation button, and complete lack of legacy downloads (bizarre) it seems like this project has been parked by someone who doesn't intend to actively develop or maintain it.

**Comment by gingerbeardman** [comment]
Any thoughts @markcarver @mstratman ?

**Comment by markcarver** [comment]
**TL;DR**

If you're just after the file and don't care about FOSS communities (as seems the case from the above comment), just click here https://slack-files.com/T7HLMPFC2-F91NUPD5J-843dfef276

---

It hasn't been parked... it is, and always has been, a side project by the original author and myself. I have and still want to keep this alive, but alas... this is not my day job (which comes first and has been quite busy lately).

I am currently the only person doing the actually "work" on this project. I cannot do absolutely everything by myself on your timetable just because _you_ want it.

This is open source.

Which means: sometimes developers get bogged down and if you want viable working software, you sometimes need to jump in yourself and help out. All the source code is right here in the repo, free and publically available.

You or others are more than welcome to open PRs to help improve the software and get it to a "stable" state.

No one is preventing you or anyone else from doing that.

---

Re: legacy downloads, this was, in fact, a by-product of the transfer to an org and semi-accident when attempting to reorganize some things (GitHub removed support for manually uploaded files, which these originally were https://blog.github.com/2012-12-12-goodbye-uploads/).

Now, GitHub's UI only allows official releases to be made that can also be linked to a specific git ref in the repository.

Given that the history of the so-called "3.0.0-betaX" releases wasn't actually 3.x code, the subsequent tags were removed in order to avoid further confusion.

If you want to download the last "semi-official" release, 2.3.7 (formerly known as 3.0.0-beta7), you can join us here https://slack.cocoadialog.com and then download the binary pinned there.

If you or someone has the last "stable" release (2.1.1), I would appreciate getting a copy, but I think it's long since been lost.

**Comment by gingerbeardman** [comment]
Here's the 2.1.1 binary

[CocoaDialog.zip](https://github.com/cocoadialog/cocoadialog/files/2271797/CocoaDialog.zip)

**Comment by markcarver** [comment]
Awesome! Thanks for that! I've gone ahead and pinned it in Slack and created an external link for it: https://slack-files.com/T7HLMPFC2-FC5BMS92R-d129c46d32

**Comment by heisian** [comment]
If you're too busy to maintain the project, it might be worthwhile to allow someone who's enthusiastic about join as a maintainer or at least state that the project isn't actively maintained, so that you don't get annoying messages like ours.

**Comment by heisian** [comment]
Also, there's this webpage: https://cocoadialog.com, that prophecies a Spring/Summer 2018 release. I suppose it's fall, now, and maybe the site should be just taken down altogether.

**Comment by markcarver** [comment]
> If you're too busy to maintain the project, it might be worthwhile to allow someone who's enthusiastic about join as a maintainer

You make this seem like I'm not enthusiastic about it.

I'm the one actually still trying to keep this project alive.

If anyone provided several PRs and proved their ability to help with the project, then that would make sense... sure. Unfortunately, no one really has and the ones that did have become unresponsive.

I'm not just going to add someone who doesn't know its history or code. That can be dangerous and confusing; as evidence from when I first joined, if I'm being honest. I didn't understand a lot about desktop software development then. Nor versioning or releases for that matter.

That is why I'm trying to consolidate all that history and start anew with better and semi-modern code (still Object-C, not Swift... that's a whole other version).

> at least state that the project isn't actively maintained, so that you don't get annoying messages like ours.

It is "active", just not on your timescale apparently. As stated above, this was originally (and still kind of is) just a **hobby project**. There is absolutely no gaurentee or warranty of its reliability or release schedule.

Once cocoadialog 3 is stable (and released) I would love to add more infrastructure to automate testing and start building a community around it. I would love to turn this into a more professional software and I would love to add more people and value to this project... it just takes people and time (value/money) to actually do all that.

Also, I don't really see them as "annoying" comments, they're just plain rude and demanding. It seems like people act like this is done by some software company that has millions and countless resources (people) to spend on its development.

That's entitlement.

No one is actually offering to help me get it out the door. That means the development becomes bottlenecked.

There's nothing I can do about that.

If you want it to go faster, then actually **get involved with its development** (like I did way back in 2011)... don't just complain about it.

> Also, there's this webpage: https://cocoadialog.com, that prophecies a Spring/Summer 2018 release. I suppose it's fall, now, and maybe the site should be just taken down altogether.

I put that up in an effort to provide some sort of announcement of the new effort (that I was doing earlier last year) and to showcase the new "brand".

It originally was fall/winter of 2017 but I had to bump it. It seems like I'll have to bump it again to later this fall/winter, such is the life of FOSS.

---

I'm really not unsympathetic to your desires. I want this out the door too. Unfortunately, y'all need a dose of reality with a project like this that hasn't seen much [real "contribution" other than a select few](https://github.com/cocoadialog/cocoadialog/graphs/contributors).

I will allocate some time each weekend after I get back from Italy later this month. How much or how long I can sustain that... remains to be determined.

**Comment by heisian** [comment]
It's fine and I wouldn't be offended if you decided to even not continue the project and ignore anyone's demands altogether. I was just suggesting saying that _perhaps_ it's worth mentioning that you're too busy to respond to simple questions or other issues.

At any rate, you produced something that people like, and we wanted an update on the status of its continuation. Nowhere in this thread did I _demand_ anything. But thank you for taking the time to respond at all.

**Comment by heisian** [comment]
This was the most inappropriate comment in the thread, IMO:
> Can we get an update on this project? Beyond the splash page and donation button, and complete lack of legacy downloads (bizarre) it seems like this project has been parked by someone who doesn't intend to actively develop or maintain it.

I agree with you, that sort of language doesn't help anything. I hope you don't feel that we're unappreciative. In my company, your package is used across our entire dev team as a build status dialog, so we use the stuff you've written all day, every day.

**Comment by markcarver** [comment]
> it's worth mentioning that you're too busy to respond to simple questions or other issues.

In all actuality, I'm usually pretty responsive to these issues because I get email notifications. Unfortunately, when I got this one from you, that was right around the time my day job started kicking backing up again and I forgot about the email and reply.

Responding to issues/comments is far easier for me than doing the actual coding which requires my local to be set up and to come up with new and non-existing code.

> At any rate, you produced something that people like, and we wanted an update on the status of its continuation.

In fairness, @mstratman is the original author and he did most of the work. I simply came in and added a few new controls that I felt were missing and made an attempt to standarize some of them as well as organize the issues 😄

> I agree with you, that sort of language doesn't help anything. I hope you don't feel that we're unappreciative.

I'm sorry if you took it personally @heisian, that wasn't my intent. I was simply triggered by that one comment and felt that a reminder needed to be made as to how to treat FOSS developers, especially ones that are essentially doing this all on their own spare time and generally for free (with a few small donations maybe here and there).

> In my company, your package is used across our entire dev team as a build status dialog, so we use the stuff you've written all day, every day.

I'm aware that it's used in this fashion. Its also used quite heavily in some JAMF/Casper deployments (which is how I became involved).

Part of the reason development has "stopped" is my work, but the other is because the last few commits dealt with some refactoring that completely broke the build and it's going to take a few nights and a lot of coffee to get it smoothed out again.

It'll happen, I just don't know an exact date.

**Comment by markcarver** [comment]
When I fix the website, I'll put an astrisk or something that has a tooltip or something explaining that it's merely an estimation... not a hard milestone.

**Comment by prohtex** [comment]
3.0b7 was released in 2013 - 5 years ago. Many apps depend on CD. The most helpful thing anyone could do at this point would be to make the documentation and download links easier to find. Put the links on the main Github page (doesn’t matter if they’re liked to the repo in releases). If it’s not feasible to fix a few bugs in 5 years time, then its time for a new maintainer to step forward. I contribute to many open source projects. It’s not about entitlement, it’s about being practical and helping support an installed user base. Branding isn’t really relevant.

**Comment by markcarver** [comment]
> 3.0b7 was released in 2013 - 5 years ago.

You know how to count. Good for you.

> Many apps depend on CD.

Not really my problem. There are other apps/programs out there that can do similar things.

> The most helpful thing anyone could do at this point would be to make the documentation and download links easier to find.

I will certainly keep that in mind (others have said the same).

I will also probably be redesigning https://cocoadialog.com soon anyway.

Mind you, this was a very simple site I threw toghether rather quickly. It's basic and has a link at the bottom to the old documentation, but apparently some people are having issues seeing it.

Not that you really care about the history or the fact that I'm really just one person...

> Put the links on the main Github page (doesn’t matter if they’re liked to the repo in releases).

There is, but again, was still only half done when I made the basic site and got pulled away from it (that happens sometimes in FOSS).

Actually, the whole goal of #12 and #90 is to automate the entire documentation process. This includes spitting out JSON info so the website can automatically be updated for specific versions. This way I, nor anyone else, has to manually update the site. 

> If it’s not feasible to fix a few bugs in 5 years time, then its time for a new maintainer to step forward.

See... this right here is what triggers me.

You make it seem like we can just pluck someone from somewhere that doesn't exist.

In that 5 years time, do you know how many people have actually **contributed** (real code) to this project... [only a very select few](https://github.com/cocoadialog/cocoadialog/pulls?q=is%3Apr+is%3Aclosed).

You make it sound as if there's some sort of backlog of PRs and other contributers are getting upset... there's not. Stop being an asshole please.

> I contribute to many open source projects.

Really doesn't sound like it. Sounds more like you just use free software and then complain when things don't go your way.

> It’s not about entitlement, it’s about being practical and helping support an installed user base.

Do you even read? Because you just defined the very definition of entitlement in the same sentence.

> Branding isn’t really relevant.

Maybe not to you. To me, its everything with this project. It used to do quite lot (and some want it to do even more). The problem with that was primarily due to not having a clear definition of _what_ its entire purpose actually was.

The core of the "app" is to make dialogs.

It's why I removed the auto updating feature. It's also why I removed the notifications.

An identity (branding) helps keep focus on these kinds of things. Its also helped with ideas of how I envision making this more of a professional solution.

I also really just hated like the icon I "designed" 7 years ago. So forgive me for being human.

---

I'm sorry @heisian and @gingerbeardman for being so late in answering your questions. Hopefully you'll see some more progress soon. I strongly recommend joining our Slack as that's where the majority of conversation will likely happen in the future.

That being said, I've answered the question and I'm locking this thread. I can't deal with anymore attacks from someone who clearly doesn't get it and just wants their voice to be heard.
